ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSliderButton.kt

```diff
@@ -0,0 +1,151 @@
+
+class UiKitSliderButton @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : ConstraintLayout(context, attrs, defStyleAttr) {
+
+    var onSlideCompleteListener: (() -> Unit)? = null
+
+    private val binding = inflateViewBinding<WidgetSliderButtonBinding>()
+
+    init {
+        context.obtainStyledAttributes(attrs, R.styleable.UiKitSliderButton).use { typedArray ->
+            val icon = typedArray.getResourceId(R.styleable.UiKitSliderButton_sliderIcon, -1)
+            if (icon != -1) binding.imageViewAction.setImageResource(icon)
+
+            val text = typedArray.getText(R.styleable.UiKitSliderButton_sliderText)
+            binding.textViewAction.text = text
+        }
+
+        setBackgroundResource(R.drawable.bg_night_rounded_big)
+
+        binding.containerOval.outlineProvider = ViewOutlineProvider.BACKGROUND
+        binding.containerOval.clipToOutline = true
+
+        binding.shimmerView.startShimmer()
+
+        initializeTouchListener()
+    }
+
+    override fun onDetachedFromWindow() {
+        binding.shimmerView.stopShimmer()
+        super.onDetachedFromWindow()
+    }
+
+    fun setupView(
+        @StringRes textRes: Int,
+        @DrawableRes iconRes: Int,
+        isLight: Boolean
+    ) {
+        with(binding) {
+            imageViewAction
+            textViewAction.setText(textRes)
+            imageViewAction.setImageResource(iconRes)
+            setLight(isLight)
+        }
+    }
+
+    fun setLight(isLight: Boolean) {
+        val nightColor = getColor(R.color.night)
+        val limeColor = getColor(R.color.lime)
+        val whiteColor = getColor(R.color.snow)
+
+        with(binding) {
+            if (isLight) {
+                setBackgroundResource(R.drawable.bg_snow_rounded_big)
+                imageViewAction.setBackgroundColor(nightColor)
+                imageViewAction.setColorFilter(limeColor, PorterDuff.Mode.SRC_IN)
+                textViewAction.setTextColor(nightColor)
+            } else {
+                setBackgroundResource(R.drawable.bg_night_rounded_big)
+                imageViewAction.setBackgroundColor(limeColor)
+                imageViewAction.setColorFilter(nightColor, PorterDuff.Mode.SRC_IN)
+                textViewAction.setTextColor(whiteColor)
+            }
+        }
+    }
+
+    @SuppressLint("ClickableViewAccessibility")
+    private fun initializeTouchListener() {
+        val initialPosition = dip(MARGIN_HORIZONTAL_DP).toFloat()
+
+        with(binding) {
+            containerOval.setOnTouchListener { view, event ->
+                return@setOnTouchListener when (event.action) {
+                    MotionEvent.ACTION_DOWN -> {
+                        setGradientVisible(isVisible = true)
+                        true
+                    }
+                    MotionEvent.ACTION_MOVE -> {
+                        val lastPosition = measuredWidth - view.width - initialPosition
```

Comment:
how about extracting all the branches to corresponding methods?

Reply:
Didn't get your idea